bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,310 @@
+    # Write the keys back to the file, for cleanliness
+    with open(S3_BUCKET_RAW_KEYS_FILE, "w") as f_obj:
+        for line in s3_bucket_raw_keys_list:
+            f_obj.write(f"{line}\n")
+
+    # Get the video files info from NZSL Signbank
+    print(
+        f"Getting raw list of video file info from NZSL Signbank ({NZSL_APP}) ...",
+        file=sys.stderr,
+    )
+    with open(NZSL_POSTGRES_RAW_KEYS_FILE, "w") as f_obj:
+        result = subprocess.run(
+            [
+                PGCLIENT,
+                "-t",
+                "-c",
+                "select id as db_id, gloss_id, is_public, videofile from video_glossvideo",
+                f"{DATABASE_URL}",
+            ],
+            env=new_env,
+            shell=False,
+            check=True,
+            text=True,
+            stdout=f_obj,
+        )
+    with open(NZSL_POSTGRES_RAW_KEYS_FILE, "r") as f_obj:
+        nzsl_raw_keys_list = f_obj.readlines()
+    print(
+        f"{len(nzsl_raw_keys_list)} rows retrieved: {NZSL_POSTGRES_RAW_KEYS_FILE}",
+        file=sys.stderr,
+    )
+
+    # Separate out the NZSL db columns
+    # Write them to a dictionary, so we can do fast operations
+    for rawl in nzsl_raw_keys_list:
+        rawl = rawl.strip()
+        if not rawl:
+            continue
+        columns = rawl.split("|")
+        db_id = columns[0].strip()
+        gloss_id = columns[1].strip()
+        is_public = columns[2].strip().lower() == "t"
+        # 'videofile' data is also the key for S3
+        video_key = columns[3].strip()
+        # Each dictionary slot contains these values
+        nzsl_raw_keys_dict[video_key] = [db_id, gloss_id, is_public]
+
+    # Get the s3 keys present and absent from our NZSL keys
+    print("Getting S3 keys present and absent from NZSL Signbank ...", file=sys.stderr)
+    for video_key in s3_bucket_raw_keys_list:
+        if video_key in nzsl_raw_keys_dict:
+            nkeys_present += 1
+            # Add 'Present' column to start
+            all_keys_dict[video_key] = [True] + nzsl_raw_keys_dict[video_key]
+        else:
+            nkeys_absent += 1
+            # Add 'Present' (absent) column to start
+            all_keys_dict[video_key] = [False, "", "", ""]
+    print(f"PRESENT: {nkeys_present} keys", file=sys.stderr)
+    print(f"ABSENT:  {nkeys_absent} keys", file=sys.stderr)
+
+    # Write all keys back to a file
+    with open(ALL_KEYS_FILE, "w") as f_obj:
+        for video_key, item_list in all_keys_dict.items():
+            outstr = (
+                f"{video_key}{CSV_DELIMITER}{CSV_DELIMITER.join(map(str, item_list))}\n"
+            )
+            f_obj.write(outstr)
+
+# From the keys present in NZSL, get all their ACL information
+print(f"Getting ACLs for keys from S3 ({AWS_S3_BUCKET}) ...", file=sys.stderr)
+# CSV header
+print(
+    f"Key{CSV_DELIMITER}Present{CSV_DELIMITER}db_id{CSV_DELIMITER}gloss_id{CSV_DELIMITER}"
+    "Public{CSV_DELIMITER}Expected{CSV_DELIMITER}Got{CSV_DELIMITER}Match"
```

Comment:
issue: can we use some more descriptive names?

I'm fine with them being short instead of verbose, but I think they could do with an extra word or two - I think probably the best question to keep in mind is "is it obvious what value we want this to be, just from the csv alone?"

e.g.
  - "Present": what's present? the file in the bucket? the value in the database? for some rows this is false whereas for others its true; which value is ideal?
    - this one might be the hardest to name succinctly; feel free to leave it as-is and we can discuss it together once we've gotten other stuff cleaned up
  - "Expected": what is expected?
  - "Got": what did we got?
  - "Match": what did we match?

Your "ID" rows (which includes "Key") are fine because they include the "what" - "gloss_id" is the id of the glossary in the database, "db_id" is... something different from "gloss_id"? but still, I think think it's better than the others I've named 😄

Reply:
Database ID is the Postgres row ID, Gloss ID is a Signbank specific ID.

Reply:
I'll groove-up the names :D

Reply:
Done.